### DISCUSSION TITLE
Different result of mp4 and mp3 input

### DISCUSSION THREAD

**Discussion by haakonjacobsen** [openai/whisper#521] [Q&A] [answered]
I'm playing around with whisper in python and found that providing an mp4 as input gave different results than providing an extracted mp3.

```python
result = model.transcribe("video/test.mp4", beam_size=5)
```
Dose not equal
```python
result = model.transcribe("video/test.mp3", beam_size=5)
```

This is the code I use to extract the mp3 from the mp4:
```python
import moviepy.editor as mp

def extract_audio_from_video():
    video = mp.VideoFileClip("video/test.mp4")
    audio = video.audio
    audio.write_audiofile("video/test.mp3")
```

I assume something happened to the mp3 when I extract it from the mp4, but I'm wondering if there is a way to extract it exactly the same way whisper handles it?

**Comment by jianfch** [comment, marked as answer]
It looks you might have re-encoded the audio stream. So it's essentially feeding in a different set of numbers, a different set of results is expected to come out. To isolate the audio stream without re-encoding it, you can use `-c:a copy` for ffmpeg.

**Reply by haakonjacobsen** [reply]
That makes sense, Thanks! 

I ended up using this command to extract the audio without encoding
```
ffmpeg -i test.mp4 -map 0:a -acodec copy audio.mp4
```
